src/ExecClient.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Buffers;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    public class ExecClient : IDisposable
+    {
+        private const int StdInStreamIndex = 0;
+        private const int StdOutStreamIndex = 1;
+        private const int StdErrStreamIndex = 2;
+        private const int ErrorChannelIndex = 3;
+        private const int ResizeChannelIndex = 4;
+
+        private readonly WebSocket socket;
+        private readonly CancellationTokenSource cts;
+        private Task runLoop;
+
+        public ExecClient(WebSocket socket)
+        {
+            this.socket = socket ?? throw new ArgumentNullException(nameof(socket));
+            this.cts = new CancellationTokenSource();
+        }
+
+        public event EventHandler ConnectionClosed;
+        public event EventHandler<string> StandardOutputReceived;
```

Comment:
I prefer java client style

ExecClient with 3 streams Stdin, Stdout, Stderr

event string does not seem to be friendly if you want to readline or something else